applications/asset_tracker/src/main.c

```diff
@@ -1437,7 +1437,9 @@ void main(void)
 			continue;
 		}
 
-		if (ret == 0) {
+		if ((ret == 0) &&
+		    (cloud_keepalive_time_left(cloud_backend) == 0)) {
+			LOG_DBG("cloud_ping()!");
 			cloud_ping(cloud_backend);
```

Comment:
If we fix the underlying issue of not propagating the `mqtt_ping()` return value, we could print whether or not the ping was acutally sent (not saying it should be done in this PR)

Reply:
I've been studiously avoiding making any changes to Zephyr :-)